charms/kserve-controller/tests/integration/test_charm.py

```diff
@@ -166,11 +186,18 @@ def assert_inf_svc_state():
     create_inf_svc()
     assert_inf_svc_state()
 
+    # Remove the InferenceService deployed in RawDeployment mode
+    lightkube_client.delete(
+        inference_service_resource, name=inf_svc_name, namespace=rawdeployment_mode_namespace
+    )
```

Comment:
The `cleanup_namespaces_after_execution` fixture is already removing everything, no need to cleanup here.

This comment also applies for the serverless test case.

Reply:
I had to add this in order for CI to pass test in GH runners. `cleanup_namespaces_after_execution` fixure removes namespace and it looks like it does not finish that work on time which causes next test to fail. Without explicit `delete` and `assert_deleted` CI passes in 30% of the cases. You can take a look at the runs attemptes in Actions tab.
I propose we keep explicit delete/assert.

Reply:
> cleanup_namespaces_after_execution fixure removes namespace and it looks like it does not finish that work on time which causes next test to fail. 

how is it causing the next test to fail? I believe that the next test case won't even look into this namespace.

Reply:
Ok, I see your point. Yes, for this case there is no need to cleanup.